enterprise-suite/templates/es-console.yaml

```diff
@@ -72,22 +79,100 @@ data:
     server {
       listen 8080;
       server_name localhost;
+      set $base /usr/share/nginx/html;
 
-      location / {
-        root  /usr/share/nginx/html;
-        index index.html index.htm;
-        try_files $uri$args $uri$args/ /index.html;
+      # use external resolver to lookup backends, cache for 30 seconds
+
+      resolver 127.0.0.1:5353 ipv6=off valid=30s;
+
+      set $prometheus "prometheus-server.{{.Release.Namespace}}.svc.cluster.local.";
+      set $monitorapi "es-monitor-api.{{.Release.Namespace}}.svc.cluster.local.";
+      set $grafana "grafana-server.{{.Release.Namespace}}.svc.cluster.local.";
+
+      # proxy_pass backends with dynamic base url
+      # (redirects to ' $1/' with a space prefix tells nginx to not add server and port)
+
+      sub_filter_once off;
+
+      location ~ (.*/service/prometheus)(/.*) {
+        proxy_pass http://$prometheus$2$is_args$args;
```

Comment:
Does this style work with `upstream` directives in nginx? Just thinking if we ever wanted to support keepalives or more than one backend, or the other config in `upstream`. I guess we can cross that bridge when we come to it.

Reply:
The proxy_pass feature has different modes, the lookup with resolver mode being triggered by the host being a variable. If we wanted it to go to an upstream, we'd use the upstream key here instead of `$prometheus` but leave the rest as-is.

```
     upstream prompool {
       ...
     }

      location ~ (.*/service/prometheus)(/.*) {
        proxy_pass http://prompool$2$is_args$args;
        ...
```

For upstreams, this lookup mode is triggered by the http block having a resolver defined. 🙄